File: frontend/src/types/video.ts
export type VideoStatus =
  | "pending"
  | "deferred"
  | "processing"
  | "completed"
  | "failed"
  | "manual";

export interface VideoResponse {
  id: string;
  course_id: string;
  youtube_video_id: string;
  title: string;
  position: number;
  duration_seconds: number | null;
  status: VideoStatus;
  transcript_source: "youtube_captions" | "groq_whisper" | null;
  celery_task_id: string | null;
  scheduled_for: string | null;
  error_message: string | null;
  created_at: string | null;
  updated_at: string | null;
}

File: backend/app/schemas/video.py
from enum import Enum
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class VideoStatus(str, Enum):
    PENDING = "pending"
    DEFERRED = "deferred"
    PROCESSING = "processing"
    RATE_LIMITED = "rate_limited"
    BATCH_PROCESSING = "batch_processing"
    COMPLETED = "completed"
    FAILED = "failed"
    MANUAL = "manual"


class VideoResponse(BaseModel):
    id: UUID
    course_id: UUID
    youtube_video_id: str
    title: str
    position: int
    duration_seconds: int | None
    status: str
    transcript_source: str | None
    celery_task_id: str | None
    scheduled_for: datetime | None
    error_message: str | None
    created_at: datetime | None
    updated_at: datetime | None

    model_config = ConfigDict(from_attributes=True)
